### ISSUE TITLE
Don't show in the dock or list of open applications when showing the menubar icon.

### ISSUE THREAD

**Issue by bwinton** [mozilla-mobile/mozilla-vpn-client#482]
It would be nice if, when I had the VPN app set up to show in the menu-bar, it didn't show in the dock or list of open applications… (I guess I might be asking for a toggle to show in the Dock+AppList or MenuBar?)

It's a common MacOS pattern to only have one way to access the main window.

(Suggested labels: `enhancement`, `macos`. :slightly_smiling_face:)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/VPN-448)

**Comment by chrismore** [comment]
@bwinton are you saying that when the Mozilla VPN is open, you don't see it in the dock+applist below and also don't see it in the top MenuBar? I see it in both locations when I have it open on my Mac.

**Comment by bwinton** [comment]
Not quite, I'm saying that I see it in both places, but should only see it in one or the other… (And preferably, for me, only in the menu bar. 🙂)

**Comment by jess-cook03** [comment]
Capturing as design feedback/suggestions and documenting elsewhere in backlog for future design iterations. Closing issue. Thanks for the feedback!

**Comment by bwinton** [comment]
Good to hear! Where can I track this, to see how it's coming along?

**Comment by AndyDeNike** [comment]
absolutely no need for this to be in top menu bar and in the dock.  Mullvad does not force you to have an open dock icon and Mozilla should follow suit.  
<img width="1280" alt="Screen Shot 2021-03-18 at 8 09 37 PM" src="https://user-images.githubusercontent.com/15356537/111728072-5bcb0300-8829-11eb-9099-ee114fcfd5ea.png">

**Comment by AndyDeNike** [comment]
this was enough to make me stop my subscription

**Comment by bakulf** [comment]
I vote to remove the dock icon. @jess-cook03 ?

**Comment by jess-cook03** [comment]
@bakulf I like this option you mentioned that Wireguard does:

> shows it (dock) only when the app is visible (when hidden, only the menubar is shown)

I think this functionality would be appropriate^

**Comment by paxperscientiam** [comment]
IMHO, this should be kept open. This is a very import UX feature as this is not a foreground application.

**Comment by erikdw** [comment]
@paxperscientiam : if you select `Hide Mozilla VPN` from the menu bar drop-down, then it:
* hides the Mozilla VPN App Window
* and it *also* hides the app in both the Dock and the App Switcher

![image](https://user-images.githubusercontent.com/1111441/145323161-6a8e53c1-87b2-4e16-93b4-71d2e980edeb.png)

Notably, if you just close the Mozilla VPN App Window with the red X button, the app does *not* get hidden from the Dock nor the App Switcher.  

![image](https://user-images.githubusercontent.com/1111441/145323606-542e5a85-271f-4d40-9693-6ede8cdc26d4.png)

And there seems to be a bug here, since closing the app that way _does_ change the menu bar preference toggle from `Hide Mozilla VPN` to `Show Mozilla VPN`.  

![image](https://user-images.githubusercontent.com/1111441/145323625-08136461-1895-4b9e-a1cf-245b6ab3c659.png)

i.e., if you get to this state and you want to hide Mozilla VPN in both the App Switcher and the Dock, you have to select `Show Mozilla VPN` and then `Hide Mozilla VPN` to actually fully hide it.  Closing the App Window gets you into a "partially hidden" state.  Presumably it "_just_" needs to have a call to `MacOSUtils::hideDockIcon()` when the App Windows is closed.  I assume there's some handler that is called when the red X is clicked and it could easily be added there.

**Comment by paxperscientiam** [comment]
Why was this closed? It's unresolved.

**Comment by reefdog** [comment]
+1'ing this issue.

VPN falls into the "always-open utility app" category, like a window manager or battery monitor etc., that I like to configure to only show in my menu bar and _not_ appear in the Dock or in the Command-Tab quick-switch menu.

My previous VPN app (Encrypt.me, née Cloak) offered this, as do many other utility-like apps.

Hardly the most critical feature of a VPN app and service, but it is a bit of a constant annoyance.

**Comment by data-sync-user** [comment]
➤ Andrea Marchesini commented:

Peter Benvenuto can you help me to define the correct behavior? Thanks.

**Comment by data-sync-user** [comment]
➤ Peter Benvenuto commented:

Andrea Marchesini we just discussed this in the office hours meeting, and we agreed that when the used hit the red close button it should remove off of the dock, and if they “hide” the app then it should stay on the dock.

**Comment by ryankilosky** [comment]
I came here just to +1 this. The dock icon should DISAPPEAR, but the menu bar icon should remain. Like @reefdog mentioned, this falls into the "always-open utility app" category. I want SO badly to use Mozilla's products, and they've all been great thus far. But, when paying for the VPN service so I can monetarily support Mozilla's development, the VPN app seems half-baked and contains small nuisances that one would think fall into the "why didn't they think of this" category.

**Comment by arnabwahid** [comment]
Wow, I came four years late to request this feature. It's disappointing that, though frequently asked, it has not been implemented yet.

**Comment by asudler** [comment]
Is this issue still ongoing? I'm thinking about purchasing Mozilla's VPN, but the inability to remove the app from the dock when running is frustrating
